### PR TITLE
Add a rule checking for react components test coverage

### PR DESCRIPTION
**Pull request by quesabe**

### CODE REVIEW

**Comment by quesabe** [comment]
Closes PLA-44.

**Review by gvidon — CHANGES_REQUESTED** [review]
@quesabe let's make sure that the rules does the following (according to [the slack discussion](https://ottofeller.slack.com/archives/GUAS246BU/p1638387797008200?thread_ts=1638380904.001700&cid=GUAS246BU)), check that:
* `__test__/[file name from rule config, e.g. index.tsx]` exists
* And has a `describe()` call with it.

File: src/react/__tests__/index.ts [hunk 1]
```diff
@@ -256,4 +257,82 @@ describe('React rules', () => {
...
+      componentHasTests(ruleParams('ComponentWithValidTest'))
+      expect(failMock).not.toHaveBeenCalled()
+    })
+
+    it('fails on a React component with no test file inside __tests__ folder', () => {
+      componentHasTests(ruleParams('ComponentWithoutTestFile'))
+      expect(failMock).toHaveBeenCalledTimes(1)
+    })
+
+    it('fails on a React component with a test file that has no component import', () => {
```
Comment (gvidon) [inline src/react/__tests__/index.ts:323]: @quesabe 👍

**Comment by gvidon** [comment]
Return early pla-44

**Review by gvidon — APPROVED** [review]

**Pull request merged**